agent/utils/utils.go

```diff
@@ -227,14 +229,15 @@ func ListPendingRequests(client *http.Client, proxyHost, backendID string) ([]st
 		return nil, fmt.Errorf("A proxy request failed: %q", err.Error())
 	}
 	defer proxyResp.Body.Close()
-	return parseRequestIDs(proxyResp)
+	return parseRequestIDs(proxyResp, metricHandler)
 }
 
-func parseRequestFromProxyResponse(backendID, requestID string, proxyResp *http.Response) (*ForwardedRequest, error) {
+func parseRequestFromProxyResponse(backendID, requestID string, proxyResp *http.Response, metricHandler *metrics.MetricHandler) (*ForwardedRequest, error) {
 	user := proxyResp.Header.Get(HeaderUserID)
 	startTimeStr := proxyResp.Header.Get(HeaderRequestStartTime)
 
 	if proxyResp.StatusCode != http.StatusOK {
+		metricHandler.WriteResponseCodeMetric(proxyResp.StatusCode)
```

Comment:
Same concern about emitting metrics for internal agent requests. This one is a bit more defensible since the individual read requests should at least correspond 1:1 with user requests.

Reply:
What's the implication of failing to forward a single request? If there's a retry mechanism, then it's probably safe to omit the call here. However, if this can lead to a scenario where user requests aren't being processed, I think it may be worth keeping. WDYT?

Reply:
We don't have a retry mechanism, so we should probably keep this one. Failing a request here would mean that a user request also fails.